tests/SimplifyTest.elm

```diff
@@ -13944,6 +13945,193 @@ a = identity
+            \() ->
+                """module A exposing (..)
+import Array
+a = Array.map f <| Array.empty
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "Array.map on Array.empty will result in Array.empty"
+                            , details = [ "You can replace this call by Array.empty." ]
+                            , under = "Array.map"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+import Array
+a = Array.empty
+"""
+                        ]
+        , test "should replace Array.empty |> Array.map f by Array.empty" <|
+            \() ->
+                """module A exposing (..)
+import Array
+a = Array.empty |> Array.map f
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "Array.map on Array.empty will result in Array.empty"
+                            , details = [ "You can replace this call by Array.empty." ]
+                            , under = "Array.map"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+import Array
+a = Array.empty
+"""
+                        ]
+        , test "should replace Array.map identity x by x" <|
+            \() ->
+                """module A exposing (..)
+a = Array.map identity x
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "Array.map with an identity function will always return the same given array"
+                            , details = [ "You can replace this call by the array itself." ]
+                            , under = "Array.map"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a = x
+"""
+                        ]
+        , test "should replace Array.map identity <| x by x" <|
+            \() ->
+                """module A exposing (..)
+import Array
+a = Array.map identity <| x
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "Array.map with an identity function will always return the same given array"
+                            , details = [ "You can replace this call by the array itself." ]
+                            , under = "Array.map"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+import Array
+a = x
+"""
+                        ]
+        , test "should replace x |> Array.map identity by x" <|
```

Comment:
For all those `x` variables I'd prefer `array` but it doesn't matter much

Reply:
(Let me know if you'd prefer me avoiding to force-push in the future)